src/common/libraries/MessageLib.sol

```diff
@@ -78,76 +78,77 @@ library MessageLib {
-        (9   << uint8(MessageType.UpdateJournal) * 8) +
-        (107 << uint8(MessageType.TriggerUpdateHoldingAmount) * 8) +
-        (91  << uint8(MessageType.TriggerUpdateShares) * 8);
+    /// @dev Each enum value corresponds to an array index holding the fixed length of the message.
+    ///      If a message has dynamic length, the value represents its base size (minimum static part).
+    function _messageLengths(uint16 kind) private pure returns (uint16) {
+        if (kind == uint16(MessageType.Invalid)) return 0; // Invalid placeholder
+        if (kind == uint16(MessageType.MessageProof)) return 34;
+        if (kind == uint16(MessageType.InitiateMessageRecovery)) return 68;
+        if (kind == uint16(MessageType.DisputeMessageRecovery)) return 68;
+        if (kind == uint16(MessageType.ScheduleUpgrade)) return 34;
+        if (kind == uint16(MessageType.CancelUpgrade)) return 34;
+        if (kind == uint16(MessageType.RecoverTokens)) return 162;
+        if (kind == uint16(MessageType.RegisterAsset)) return 179;
+        if (kind == uint16(MessageType.NotifyPool)) return 10;
+        if (kind == uint16(MessageType.NotifyShareClass)) return 251;
+        if (kind == uint16(MessageType.UpdateShareClassPrice)) return 66;
+        if (kind == uint16(MessageType.UpdateShareClassMetadata)) return 186;
+        if (kind == uint16(MessageType.UpdateShareClassHook)) return 58;
+        if (kind == uint16(MessageType.TransferShares)) return 74;
+        if (kind == uint16(MessageType.UpdateRestriction)) return 26; // dynamic
+        if (kind == uint16(MessageType.UpdateContract)) return 58; // dynamic
+        if (kind == uint16(MessageType.DepositRequest)) return 90;
+        if (kind == uint16(MessageType.RedeemRequest)) return 90;
+        if (kind == uint16(MessageType.FulfilledDepositRequest)) return 106;
+        if (kind == uint16(MessageType.FulfilledRedeemRequest)) return 106;
+        if (kind == uint16(MessageType.CancelDepositRequest)) return 74;
+        if (kind == uint16(MessageType.CancelRedeemRequest)) return 74;
+        if (kind == uint16(MessageType.FulfilledCancelDepositRequest)) return 90;
+        if (kind == uint16(MessageType.FulfilledCancelRedeemRequest)) return 90;
+        if (kind == uint16(MessageType.TriggerRedeemRequest)) return 90;
+        if (kind == uint16(MessageType.UpdateHoldingAmount)) return 115; // dynamic
+        if (kind == uint16(MessageType.UpdateHoldingValue)) return 66;
+        if (kind == uint16(MessageType.UpdateShares)) return 99;
+        // if (kind == uint16(MessageType.ApprovedDeposits)) return 58;
+        // if (kind == uint16(MessageType.RevokedShares)) return 58;
+        if (kind == uint16(MessageType.UpdateJournal)) return 10; // dynamic
+        if (kind == uint16(MessageType.TriggerUpdateHoldingAmount)) return 108; // dynamic
+        if (kind == uint16(MessageType.TriggerUpdateShares)) return 92;
+        revert UnknownMessageType();
```

Comment:
The problem with this is that it's much more slow if the messages are the last ones in the list